### PR TITLE
Use pending nonce over true nonce by default for txs

### PR DESCRIPTION
**Pull request by Daniel-VDM**
Using the pending nonce allows for fast consecutive transfers when using the batch transaction file or if using `--timeout 0` flag. However, should one want to use the true nonce (from finalized transactions, i.e: the old way) the `--true-nonce` option has been added to do just that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,106 +47,84 @@ Note:
...
-hmy --node="https://api.s0.t.hmny.io" blockchain utility-metrics
+./hmy --node=https://api.s0.t.hmny.io blockchain utility-metrics
 
 19. Check in-memory record of failed staking transactions
-hmy failures staking
+./hmy --node=https://api.s0.t.hmny.io failures staking
 
 20. Check which shard your BLS public key would be assigned to as a validator
-hmy utility shard-for-bls 2d61379e44a772e5757e27ee2b3874254f56073e6bd226eb8b160371cc3c18b8c4977bd3dcb71fd57dc62bf0e143fd08
+./hmy --node=https://api.s0.t.hmny.io utility shard-for-bls 2d61379e44a772e5757e27ee2b3874254f56073e6bd226eb8b160371cc3c18b8c4977bd3dcb71fd57dc62bf0e143fd08
```
Comment (janet-harmony) [inline README.md:127]: `<BLS_PUBLIC_KEY>`?

File: README.md [hunk 2]
```diff
@@ -178,6 +156,7 @@ The JSON file will be a JSON array where each element has the following attribut
 | `gas-price`         | string     | [*Optional*] The gas price to pay in NANO (1e-9 of $ONE), default is 1. |
 | `gas-limit`         | string     | [*Optional*] The gas limit, default is 21000. |
 | `stop-on-error`     | boolean    | [*Optional*] If true, stop sending transactions if an error occurred, default is false. |
+| `true-nonce`        | boolean    | [*Optional*] If true, send transaction using true on-chain nonce. Cannot be used with `nonce`. If non is provided, use tx pool nonce. |
```
Comment (janet-harmony) [inline README.md:159]: None?

File: cmd/subcommands/staking.go [hunk 3]
```diff
@@ -388,9 +388,14 @@ Create a new validator"
 				}
 			}
 
-			nonce, err := getNonceFromInput(validatorAddress.String(), inputNonce, networkHandler)
-			if err != nil {
-				return err
+			var nonce uint64
```
Comment (janet-harmony) [inline cmd/subcommands/staking.go:391]: Can probably use a `PreRunE` function, instead of duplicating this many times

File: README.md (hunk 1 above)
Comment (Daniel-VDM) [reply README.md]: updated cookbook doc

File: cmd/subcommands/staking.go (hunk 3 above)
Comment (Daniel-VDM) [reply cmd/subcommands/staking.go]: prerun prob messier

**Pull request merged**